cmd/agent/app/reporter/reporter.go

```diff
@@ -26,6 +26,7 @@ import (
 type Reporter interface {
 	EmitZipkinBatch(spans []*zipkincore.Span) (err error)
 	EmitBatch(batch *jaeger.Batch) (err error)
+	Close() error
```

Comment:
Another example that causes cascading changes in this PR without any apparent benefits, since the two main reporters (grpc and tchannel) have no-op Close().